src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php

```diff
@@ -510,6 +515,11 @@ static function (): bool {
                     }
                 );
             }
+
+            $this->lastActivityTime = time();
+        } elseif (!empty($this->connectionOptions['heartbeat']) && time() > $this->lastActivityTime + ($this->connectionOptions['heartbeat'] * 2)) {
```

Comment:
`} elseif (0 < ($this->connectionOptions['heartbeat'] ?? 0) && time() > $this->lastActivityTime + 2 * $this->connectionOptions['heartbeat']) {`